.github/workflows/reusable-build-docker-image.yaml

```diff
@@ -0,0 +1,221 @@
+name: Build docker image
+
+on:
+  # Reusable workflow
+  workflow_call:
+    secrets:
+      token:
+        required: true
+      docker_token:
+        required: true
+
+# Assuming the below is in the workflow for an individual repository.
+# on:
+#   # triggers the workflow on published release
+#   release:
+#     types:
+#       - published
+#   # allows run of this workflow manually from the actions tab
+#   # must be merged to default before it will be available to manually run.
+#   workflow_dispatch:
+
+env:
+  # REQUIRED organization level variables used in this workflow.
+  #   ${{ github.event.repository.default_branch }} is used throughout
+  #     rather than main to generalize default branch
+  #     (removed DEFAULT_BRANCH org var)
+  #   DOCKER_REGISTRY: URL to the docker registry
+  #   DOCKER_REGISTRY_USER: user to sign into DOCKER_REGISTRY
+  #   DOCKER_REGISTRY_TOKEN: token for USER user to sign into DOCKER_REGISTRY
+  #   RUNNER: default runner
+  #   UPLOAD_GITHUB_ARTIFACTS: true/false, specify whether to upload artifacts to github
+  #   USE_DOCKER_BUILDX_ACTION: true/false, specify whether to use the docker buildx action
+
+  # OPTIONAL organization level variables:
+  #   # Some system implementations may require different runners for building and
+  #   # running docker containers - allow specifying separately. Defaults to vars.RUNNER
+  #   # if these are not defined. Note env context nor strings are supported in the
+  #   # runs-on field, so we MUST use the vars context for the default.
+  #   RUNNER_DOCKERBUILD: runner to build docker containers
+  #   RUNNER_DOCKERRUN: runner to run docker containers
+
+  # The path to the GeoIPS package in DOCLINTTEST images
+  BASE_GEOIPS_PATH: /packages/geoips
+  # Allow overriding the doclinttest image tag used for plugin packages
+  # Allows using a different version of the doc, lint, and test code
+  PLUGIN_DOCLINTTEST_TAG: doclinttest-stable
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  # Set variables that should be accessible to all jobs
+  get-job-flags:
+    runs-on: ${{ vars.RUNNER }}
+    env:
+      CURR_REPO: ${{ github.event.repository.name }}
+    outputs:
+      # The name of the untagged docker images to be created by this workflow
+      IMAGE_NAME: ${{ steps.set_image_name.outputs.IMAGE_NAME }}
+      REPO_NAME: ${{ steps.set_image_name.outputs.REPO_NAME }}
+      # The tag to use for the doclinttest images
+      DOCLINTTEST_TAG: ${{ steps.set_doclinttest_tag.outputs.DOCLINTTEST_TAG }}
+      # The image to use for doclinttest jobs
+      DOCLINTTEST_IMAGE: ${{ steps.set_doclinttest_image.outputs.DOCLINTTEST_IMAGE }}
```

Comment:
In reusable-doc-test.yaml, this also icludes variables for requesting build docs test, test_interfaces, pytest, etc.

Since we pulled build-docker-image part out of doc-test, removed references to actual testing variables.